### PR TITLE
feat(FEC-10110): sanitize host with whitelist

### PR DESCRIPTION
**Pull request by OrenMe**
if HTTP XFORWARDED HOST header is present check it against whitelist

### CODE REVIEW

File: includes/DefaultSettings.php [hunk 1]
```diff
@@ -252,6 +259,17 @@
 	require_once( $wgLocalSettingsFile );
 }
 
+//If http host is not in whitelist reset config to original request host
+if (!in_array($_SERVER['HTTP_HOST'], $wgRemoteAddrWhitelistedHosts )) {
+    $_SERVER["HTTP_HOST"] = $origHttpHost;
```
Comment (gonenradai) [inline includes/DefaultSettings.php:264]: maybe change to single quote here too, for consistency

Comment (OrenMe) [reply includes/DefaultSettings.php]: probably copy pasted it before I changed it :-)

File: includes/DefaultSettings.php [hunk 2]
```diff
@@ -252,6 +258,20 @@
 	require_once( $wgLocalSettingsFile );
 }
 
+//If http host is not in whitelist reset config to original request host
+if ((isset($_SERVER["HTTP_X_FORWARDED_HOST"]) &&
+    !in_array($_SERVER['HTTP_HOST'], $wgRemoteAddrWhitelistedHosts )) ||
```
Comment (gonenradai) [inline includes/DefaultSettings.php:263]: shouldn't this be HTTP_X_FORWARDED_HOST?

**Comment by yairans** [comment]
resolved by https://github.com/kaltura/mwEmbed/pull/4266

**Pull request closed without merging**